code/modules/mob/living/simple_animal/abnormality/_tools/he/fateloom.dm

```diff
@@ -0,0 +1,68 @@
+#define STATUS_EFFECT_REDSTRING /datum/status_effect/stacking/red_string
+/obj/structure/toolabnormality/fateloom
+	name = "third fate's loom"
+	desc = "It is surrounded by spools of red thread."
+	icon_state = "loom"
+	var/usage_cooldown
+	var/usage_cooldown_time = 5 SECONDS
+
+/obj/structure/toolabnormality/fateloom/attack_hand(mob/living/carbon/human/user)
+	. = ..()
+	if(!do_after(user, 10))
+		return
+	if(usage_cooldown > world.time) //just to prevent sfx spam
+		to_chat(user, "<span class='warning'>The loom is alraedy spinning!</span>")
+		return
+	usage_cooldown = world.time + usage_cooldown_time
+
+	var/datum/status_effect/stacking/red_string/S = user.has_status_effect(/datum/status_effect/stacking/red_string)
+	if(!S)
+		to_chat(user, "<span class='userdanger'>As you touch the loom, threads are sewn into your flesh.</span>")
+		user.apply_status_effect(STATUS_EFFECT_REDSTRING)
+	else
+		to_chat(user, "<span class='userdanger'>The threads which were once sparse are now reinforced.</span>")
+		to_chat(user, "<span class='userdanger'>You feel weaker.</span>")
+		S.add_stacks(4)
+		user.adjust_attribute_level(FORTITUDE_ATTRIBUTE, -15)
+	playsound(src, 'sound/abnormalities/fateloom/garrote_bloody.ogg', 80, TRUE, -3)
+
+// Status Effect
+/datum/status_effect/stacking/red_string
+	id = "stacking_red_string"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = -1
+	alert_type = null
+	stack_decay = 0
+	stacks = 4 //was 3, until I figured out that 0 stacks causes stacking status effects to forcibly qdel
+	max_stacks = 4
+	consumed_on_threshold = FALSE
+
+/datum/status_effect/stacking/red_string/on_apply()
+	RegisterSignal(owner, COMSIG_MOB_APPLY_DAMGE, .proc/heal)
+	return ..()
+
+/datum/status_effect/stacking/red_string/proc/heal()
+	SIGNAL_HANDLER
+	if(owner.health <= owner.maxHealth * 0.5)
+		var/mob/living/carbon/human/H = owner
+		if(stacks > 1)
+			playsound(H, 'sound/abnormalities/fateloom/garrote.ogg', 80, TRUE, -3)
+			H.adjustBruteLoss(-(H.maxHealth * 0.5))
+			H.adjustSanityLoss(H.maxHealth * 0.5) // lose sanity by how much health you gain
+			if(stacks > 2)
+				to_chat(H, "<span class='userdanger'>You lose some of your threads!</span>")
+			else if (stacks == 2)
+				to_chat(H, "<span class='userdanger'>You are running low on threads!</span>")
+		else
+			to_chat(H, "<span class='userdanger'>Your entire body falls apart!</span>")
+			for(var/X in H.bodyparts)
+				var/obj/item/bodypart/BP = X
+				if(BP.body_part && BP.body_part != CHEST)
+					if(BP.dismemberable)
+						BP.dismember()
+			playsound(H, 'sound/abnormalities/fateloom/garrote_bloody.ogg', 80, TRUE, -3)
+			new /obj/effect/gibspawner/generic/silent(get_turf(H))
+			H.regenerate_icons()
+		stacks -= 1
```

Comment:
It is preferred that you use early returns in such cases.

I assume we should be deleting the status effect after it dismembers the user, isn't that so? Otherwise it could result in a loop of trying to dismember limbs that aren't here anymore.

Also, you might want to kill the user after dismemberment. As funny as it is, I don't think anyone will enjoy being a nugget that can't even move on their own(isn't included in the suggestion).

Also added a (potentially obsolete) stat and QDELETED check just so it doesn't run when it shouldn't
```suggestion
/datum/status_effect/stacking/red_string/proc/heal()
	SIGNAL_HANDLER
	var/mob/living/carbon/human/H = owner
	if(QDELETED(H) || H.stat == DEAD)
		return
	if(H.health > H.maxHealth * 0.5)
		return

	if(stacks > 1)
		playsound(H, 'sound/abnormalities/fateloom/garrote.ogg', 80, TRUE, -3)
		H.adjustBruteLoss(-(H.maxHealth * 0.5))
		H.adjustSanityLoss(H.maxHealth * 0.5) // lose sanity by how much health you gain
		if(stacks > 2)
			to_chat(H, "<span class='userdanger'>You lose some of your threads!</span>")
		else if(stacks == 2)
			to_chat(H, "<span class='userdanger'>You are running low on threads!</span>")
		stacks -= 1
		return

	to_chat(H, "<span class='userdanger'>Your entire body falls apart!</span>")
	for(var/X in H.bodyparts)
		var/obj/item/bodypart/BP = X
		if(BP.body_part && BP.body_part != CHEST)
			if(BP.dismemberable)
				BP.dismember()
	playsound(H, 'sound/abnormalities/fateloom/garrote_bloody.ogg', 80, TRUE, -3)
	new /obj/effect/gibspawner/generic/silent(get_turf(H))
	H.regenerate_icons()
	QDEL_NULL(src)
```

Reply:
> It is preferred that you use early returns in such cases.
> 
> I assume we should be deleting the status effect after it dismembers the user, isn't that so? Otherwise it could result in a loop of trying to dismember limbs that aren't here anymore.
> 
> Also added a (potentially obsolete) stat and QDELETED check just so it doesn't run when it shouldn't
1. Ah, forgot to implement this after removing signals. Good catch.

2. Reducing stacks to 0 removes the status effect automatically, handled in stacking status_effects.

3. Check is probably obsolete but doesn't hurt not to see runtimes in the future. 

Tested the suggestion and everything works fine, so I'm implementing it.